### DISCUSSION TITLE
How to Email OTP?

### DISCUSSION THREAD

**Discussion by ioucyf** [supabase/supabase#11840] [Questions] [answered]
Hello, 

I'd like to sign-in (or sign-up if new) users with email and OTP. It doesn't explain how to set that up in the documentation.

is it possible? or is OTP only available for mobile phones? 

if possible, do I have to generate the OTP codes myself? or does it take care of generating it for me? 

so far, that's what I have for sign in with email OTP: 
```
// always sends a magic link :(
// I don't see an optional field to allow me to choose between otp or magiclink

const { error, data } = await client.auth.signInWithOtp({
      email: 'example@email.com', // required field for my setup
      options: {
        shouldCreateUser: true,
     // data: 'i don't know what goes here',
     // captchaToken: 'i don't know what goes here',
      }
    });
if (error) throw error;
```

then there's the verifyOtp method: 
```
// email, token, type are required 
// on type option for 'signin'

  const { error, data } = await client.auth.verifyOtp({
      email: 'example@email.com',
   // token: 'is this to verify the submitted otp?',
   // type: 'magiclink' | 'signup' | 'invite' | 'recovery' | 'email_change' | 'phone_change' | 'sms'
    });
if (error) throw error;
```

I couldn't find anything int he docs to explain in details what goes where and how these two methods work and/or should be used. 

Thank you for your time.

**Comment by GaryAustin1** [comment, marked as answer]
To send the OTP you just put {{.Token}} in the email template for the operation you are confirming. Then in your code wait in a screen to get the input (6 chars by default).   Call verifyOtp with the code and type of token.

**Reply by ioucyf** [reply]
Thank you very much!
